appveyor/numpy.patch

```diff
@@ -0,0 +1,831 @@
+             if requiref90:
+                 if fcompiler.module_dir_switch is None:
+                     existing_modules = glob('*.mod')
+-                extra_postargs += fcompiler.module_options(\
++                extra_postargs += fcompiler.module_options(
+                     module_dirs, module_build_dir)
+ 
+             if fmodule_sources:
+@@ -257,14 +298,14 @@ class build_clib(old_build_clib):
+                     if f in existing_modules:
+                         continue
+                     t = os.path.join(module_build_dir, f)
+-                    if os.path.abspath(f)==os.path.abspath(t):
++                    if os.path.abspath(f) == os.path.abspath(t):
+                         continue
+                     if os.path.isfile(t):
+                         os.remove(t)
+                     try:
+                         self.move_file(f, module_build_dir)
+                     except DistutilsFileError:
+-                        log.warn('failed to move %r to %r' \
++                        log.warn('failed to move %r to %r'
+                                  % (f, module_build_dir))
+ 
+             if f_sources:
+@@ -278,11 +319,49 @@ class build_clib(old_build_clib):
+         else:
+             f_objects = []
+ 
+-        objects.extend(f_objects)
+-
+-        # assume that default linker is suitable for
+-        # linking Fortran object files
+-        compiler.create_static_lib(objects, lib_name,
++        libs = []
++
++        if objects:
++            # Go ahead and compile the c code
++            compiler.create_static_lib(objects, lib_name + '_msvc',
++                                       output_dir=self.build_clib,
++                                       debug=self.debug)
++            libs += [os.path.join(self.build_clib, lib_name + '_msvc.lib')]
++
++        if f_objects:
++            if lib_name == 'odepack':
++                f_objects = [
++                    obj for obj in f_objects if 'xsetf' not in obj and 'xsetun' not in obj]
++
++            # link the fortran sources with gfortran
++            fcompiler.link_shared_object(f_objects, lib_name + '_gfortran.dll',
++                                         output_dir=self.build_clib,
++                                         library_dirs=[
++                                             self.build_clib,
++                                             '.',
++                                         ],
++                                         extra_postargs=self.dlls + [
++                                             '-Wl,--out-implib,' + lib_name + '.dll.a',
++                                             '-Wl,--output-def,' + lib_name + '_gfortran.def',
++                                             '-Wl,--export-all-symbols',
++                                             '-Wl,--enable-auto-import',
++                                         ],
++                                         debug=self.debug)
++
```

Comment:
Here we may be building C, fortran, or both. If we need to build fortran, then we need to generate a fortran DLL and a shared lib (see numpy/numpy#9427)